sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExtractValue.scala

```diff
@@ -195,6 +205,11 @@ case class GetArrayItem(child: Expression, ordinal: Expression)
       baseValue(index)
     }
   }
+
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    defineCodeGen(ctx, ev,
+      (eval1, eval2) => s"(${ctx.javaType(dataType)})$eval1.apply((int)$eval2)")
```

Comment:
According to http://stackoverflow.com/questions/12381190/drools-changing-the-java-compiler-level, looks like Janino doesn't support generic...